tests/index.test.ts

```diff
@@ -494,21 +494,45 @@ describe('PTY', { repeats: 500 }, () => {
   });
 });
 
-describe('cgroup opts', () => {
+describe('cgroup opts', async () => {
+  let SLICE_DIR: string;
+  let ORIGINAL_CGROUP: string;
+  let SLICE: string;
+  if (!IS_DARWIN) {
+    const CG_ROOT = '/sys/fs/cgroup';
+    // unique slice name to avoid conflicts with other test runs
+    SLICE = `ruspty-${Math.random().toString(36).substring(2, 15)}`;
+    SLICE_DIR = join(CG_ROOT, SLICE);
+    
+    // Get the current process's cgroup path to restore it later
+    const CGROUP_RAW = (await exec(`cat /proc/self/cgroup`)).stdout.trim();
+    // Extract just the path portion from the cgroup format (e.g., "0::/user.slice/...")
+    const CGROUP_PATH = CGROUP_RAW.split(':').pop() || '';
+    // Construct the full filesystem path to the original cgroup
+    ORIGINAL_CGROUP = join(CG_ROOT, CGROUP_PATH.replace(/^\//, ''));
```

Comment:
maybe better to set vitest to --pool=forks so we can avoid this whole set and unset cgroup thing

Reply:
tbh it's better to keep the set/unset cgroup thing. correctness ftw.